erpnext/assets/doctype/asset/test_asset.py

```diff
@@ -221,6 +221,31 @@ def test_scrap_asset(self):
 
 		scrap_asset(asset.name)
 		asset.load_from_db()
+
+		asset_depriciation = frappe.db.get_value("Asset Depreciation Schedule", asset.name, "name")
+		depreciation_entries = frappe.db.get_all(
+			"Depreciation Schedule",
+			filters={"parent": asset_depriciation},
+			fields=["schedule_date", "journal_entry"],
+			order_by="schedule_date",
+		)
+
+		last_depreciated_row = None
+		for entry in depreciation_entries:
+			if entry["journal_entry"]:
+				last_depreciated_row = entry
+
+		if asset.disposal_date > asset.purchase_date:
+			if last_depreciated_row:
+				disposal_date = frappe.utils.getdate(asset.disposal_date)
+				last_schedule_date = frappe.utils.getdate(last_depreciated_row["schedule_date"])
+				today_date = frappe.utils.getdate(nowdate())
+
+				self.assertTrue(
```

Comment:
This does not make any sense, you should check for `assertRaise` by defining an error class to it.